xrpackage.js

```diff
@@ -434,7 +436,18 @@ const xrTypeAdders = {
     const xrfb = this.realSession ? this.realSession.renderState.baseLayer.framebuffer : this.fakeXrFramebuffer;
     p.setXrFramebuffer(xrfb);
 
-    await p.context.requestPresentPromise;
+    if (options && options.timeout) {
+      const timeoutPromise = new Promise((resolve, reject) =>
+        setTimeout(() => reject(new Error('Timed out whilst loading package')), options.timeout),
```

Comment:
This feels a bit dirty, for the reason that every *successful* site package add will create this timeout in the future, which will do nothing except waste memory.

A better approach that doesn't cause a 30 second leak of the closure would be to do `clearTimeout` in the success case rather than blindly racing these promises.

Reply:
Changed it to clear the timeout whenever the original promise resolves/rejects: https://github.com/webaverse/xrpackage/pull/111/commits/b042b33f7a0c87619e5ef8240820e7e597c4bfb5

Reply:
Actually, https://github.com/webaverse/xrpackage/pull/111/commits/86c1404ddd2c1bc7145b4e2b30a6da8920bbde97 this one!